core/consensus/babe/src/lib.rs

```diff
@@ -75,28 +155,920 @@ impl Decode for BabeSeal {
 	}
 }
 
+/// A slot duration. Create with `get_or_compute`.
+// FIXME: Once Rust has higher-kinded types, the duplication between this
+// and `super::babe::Config` can be eliminated.
+pub struct Config(slots::SlotDuration<BabeConfiguration>);
+
+impl Config {
+	/// Either fetch the slot duration from disk or compute it from the genesis
+	/// state.
+	pub fn get_or_compute<B: Block, C>(client: &C) -> CResult<Self>
+	where
```

Comment:
Style: I think this `where` can be bumped up to the previous line or should be indented.